src/main/java/com/google/sps/servlets/ShowInterviewersServlet.java

```diff
@@ -0,0 +1,162 @@
+    this.personDao = personDao;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String utc = request.getParameter("utc");
+    Instant startOfRange = Instant.parse(utc);
+    Instant endOfRange = Instant.parse(utc).plus(1, ChronoUnit.HOURS);
+    List<Availability> availabilitiesInRange =
+        availabilityDao.getInRangeForAll(startOfRange, endOfRange);
+    List<Person> possiblePeople =
+        getPossiblePeople(availabilitiesInRange, startOfRange, endOfRange);
+    Set<PossibleInterviewer> possibleInterviewers = getPossibleInterviewers(possiblePeople);
+    request.setAttribute("interviewers", possibleInterviewers);
+    RequestDispatcher rd = request.getRequestDispatcher("/possibleInterviewers.jsp");
+    try {
+      rd.forward(request, response);
+    } catch (ServletException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  List<Person> getPossiblePeople(
+      List<Availability> allAvailabilities, Instant startOfRange, Instant endOfRange) {
+    Set<String> allInterviewers = new HashSet<String>();
+    for (Availability avail : allAvailabilities) {
+      allInterviewers.add(avail.userId());
+    }
+
+    // We don't want to schedule an interview for a user with themself, so we are removing
+    // the current user's id from the list.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    String userId = userService.getCurrentUser().getUserId();
+
+    // Since UserId does not have a valid Mock, if the id is null (as when testing), it will be
+    // replaced with this hashcode.
+    if (userId == null) {
+      userId = String.format("%d", userEmail.hashCode());
+    }
+
+    allInterviewers.remove(userId);
+
+    List<Person> possibleInterviewers = new ArrayList<Person>();
+    for (String interviewer : allInterviewers) {
+      if (personHasPossibleInterviewSlot(interviewer, startOfRange, endOfRange)) {
+        possibleInterviewers.add(personDao.get(interviewer).get());
+      }
+    }
+
+    return possibleInterviewers;
+  }
+
+  private Set<PossibleInterviewer> getPossibleInterviewers(List<Person> possiblePeople) {
+    Set<PossibleInterviewer> possibleInterviewers = new HashSet<PossibleInterviewer>();
+    for (Person person : possiblePeople) {
+      possibleInterviewers.add(personToPossibleInterviewer(person));
+    }
+
+    return possibleInterviewers;
+  }
+
+  boolean personHasPossibleInterviewSlot(String userId, Instant startOfRange, Instant endOfRange) {
+    List<Availability> availabilities =
+        availabilityDao.getInRangeForUser(userId, startOfRange, endOfRange);
+    List<Availability> scheduledAvailability = new ArrayList<Availability>();
+    for (Availability avail : availabilities) {
+      if (avail.scheduled()) {
+        scheduledAvailability.add(avail);
+      }
+    }
+    availabilities.removeAll(scheduledAvailability);
```

Comment:
Shorten this with `removeIf()`
https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#removeIf-java.util.function.Predicate-

Reply:
Done